application-analytics-ui/src/main/resources/Analytics/Code/Macros/VisitsSummary.xml

```diff
@@ -0,0 +1,286 @@
+      </name>
+      <parse>
+        <disabled>0</disabled>
+        <displayFormType>select</displayFormType>
+        <displayType>yesno</displayType>
+        <name>parse</name>
+        <number>4</number>
+        <prettyName>Parse content</prettyName>
+        <unmodifiable>0</unmodifiable>
+        <classType>com.xpn.xwiki.objects.classes.BooleanClass</classType>
+      </parse>
+      <use>
+        <cache>0</cache>
+        <disabled>0</disabled>
+        <displayType>select</displayType>
+        <freeText>forbidden</freeText>
+        <largeStorage>0</largeStorage>
+        <multiSelect>0</multiSelect>
+        <name>use</name>
+        <number>3</number>
+        <prettyName>Use this extension</prettyName>
+        <relationalStorage>0</relationalStorage>
+        <separator> </separator>
+        <separators>|, </separators>
+        <size>1</size>
+        <unmodifiable>0</unmodifiable>
+        <values>currentPage|onDemand|always</values>
+        <classType>com.xpn.xwiki.objects.classes.StaticListClass</classType>
+      </use>
+    </class>
+    <property>
+      <cache>long</cache>
+    </property>
+    <property>
+      <code>define('analyticsGraphMessages', {
+  prefix: 'analytics.graph.',
+  keys: [
+    'error'
+  ]
+});
+
+require.config({
+  paths: {
+    chart: "$services.webjars.url('org.webjars:chartjs', 'Chart.bundle.min.js')"
+  }
+});
+
+require(['jquery', 'xwiki-l10n!analyticsGraphMessages', 'chart'], function ($, l10n) {
+  const SELECTORS = {
+    options: '.analytics_genericVisitsFields',
+    parent: '.analytics_graphMacro',
+    canvas: 'canvas',
+    macroOptions: '.analytics_visitsGenericMacroOptions',
+    period: '.analytics_visitsGenericPeriods',
+    updateCanvas: '.analytics_visitsGenericPeriods, .analytics_genericVisitsFields'
+  };
+  const RGB = [`rgb(255, 0, 0)`, `rgb(0, 255, 0)`, `rgb(0, 0, 255)`]
+
+  // A map to store charts for dynamic updates.
+  let chartsMap = new Map();
+
+  // Listen for changes on the period or the field that is displayed and update the chart.
+  $(document).on('change', SELECTORS.updateCanvas, function () {
+    let parent = $(this).closest(SELECTORS.parent);
+    let cacheId = parent.attr('id');
+    let timeInterval = parent.find(SELECTORS.macroOptions).data('date');
+    let canvas = parent.find(SELECTORS.canvas);
+    let axis = $('option:selected', parent.find(SELECTORS.options)).data('label').split('&amp;');
+    let endpoints = $('option:selected', parent.find(SELECTORS.options)).data('endpoints').split('&amp;');
+    let fields = parent.find(SELECTORS.options).val().split('&amp;');
+    let period = parent.find(SELECTORS.period).val()
+    canvas.addClass("loading");
+    editCanvas(canvas, cacheId, timeInterval, endpoints, fields, period, axis);
+  });
+
+  // Function to create a new chart.
+  function createChart(canvas, labels, values, axis) {
+    let data = {
+      labels: labels,
+      datasets: []
+    };
+    axis.forEach((yAxeName, i) =&gt; {
```

Comment:
A forgotten `axe`

Reply:
I meant that you updated the `yAxes` name to not use `axe` (=topor), but you forgot to update `yAxeName` to not use `axe` anymore